scipy/stats/tests/test_stats.py

```diff
@@ -900,6 +900,22 @@ def test_index_error(self):
             0.0007835762419683435])
         assert_raises(ValueError, stats.spearmanr, x, y, axis=2)
 
+    def test_alternative(self):
+        x1 = [1, 2, 3, 4, 5]
+        x2 = [5, 6, 7, 8, 7]
+        expected = (0.82078268166812329, 0.088587005313543798)
+
+        assert_raises(ValueError, stats.spearmanr, X, X,
+                      alternative='error')
+
+        res = stats.spearmanr(x1, x2, alternative="less")
```

Comment:
Have you tested with the different `nan_policy`s?

Reply:
Because it was pretty easy, I added alternative to the masked variant to work out the nan_policy stuff (and wrote tests for it). As for multiple axes, it looks like `test_normal_draws` could make a good base.

Reply:
> I'll keep looking.

What's the status here?

Reply:
Sorry for the delay, multidimensional inputs seem to work fine. I've added a test for them as well as nan inputs.